Raven.Client.Lightweight/Changes/RemoteChangesClientBase.cs

```diff
@@ -201,45 +201,39 @@ private bool DisposeHeartbeatTimer()
             }
         }
 
-        private Task lastSendTask;
-
         protected Task Send(string command, string value)
         {
-            lock (this)
+            try
             {
                 logger.Info("Sending command {0} - {1} to {2} with id {3}", command, value, url, id);
-                var sendTask = lastSendTask;
-                if (sendTask != null)
+                var sendUrlBuilder = new StringBuilder();
+                sendUrlBuilder.Append(url);
+                sendUrlBuilder.Append("/changes/config?id=");
+                sendUrlBuilder.Append(id);
+                sendUrlBuilder.Append("&command=");
+                sendUrlBuilder.Append(command);
+
+                if (string.IsNullOrEmpty(value) == false)
                 {
-                    return sendTask.ContinueWith(_ =>
-                    {
-                        Send(command, value);
-                    });
+                    sendUrlBuilder.Append("&value=");
+                    sendUrlBuilder.Append(Uri.EscapeUriString(value));
                 }
 
-                try
+                var requestParams = new CreateHttpJsonRequestParams(null, sendUrlBuilder.ToString(), HttpMethods.Get, credentials, Conventions)
                 {
-                    var sendUrl = url + "/changes/config?id=" + id + "&command=" + command;
-                    if (string.IsNullOrEmpty(value) == false)
-                        sendUrl += "&value=" + Uri.EscapeUriString(value);
-
-                    var requestParams = new CreateHttpJsonRequestParams(null, sendUrl, HttpMethods.Get, credentials, Conventions)
-                    {
-                        AvoidCachingRequest = true
-                    };
-                    var request = jsonRequestFactory.CreateHttpJsonRequest(requestParams);
-                    lastSendTask = request.ExecuteRequestAsync().ObserveException();
+                    AvoidCachingRequest = true
+                };
+                var request = jsonRequestFactory.CreateHttpJsonRequest(requestParams);
+                var sendTask = request.ExecuteRequestAsync().ObserveException();
 
-                    return lastSendTask.ContinueWith(task =>
-                    {
-                        lastSendTask = null;
-                        request.Dispose();
-                    });
-                }
-                catch (Exception e)
+                return sendTask.ContinueWith(task =>
```

Comment:
Can change this to async and simplify further, no?

Reply:
Sure. I was assuming you deliberately wanted to avoid the statemachine on this path